### DISCUSSION TITLE
How are regions determined?

### DISCUSSION THREAD

**Discussion by Spectre5** [blakeblackshear/frigate#8731] [Ask A Question]
I know from the docs that motion boxes are combined to make a region, along with requiring it to be square and also adding some margin on the edges.  That all makes sense.

However, as I watch a camera in debug mode, I'll see 1 small red box (motion detected), but then the green region box will be much, much larger.  Sometimes even the entire camera frame.  All with only 1 small red box.  Is there a good way to figure out what is going on?  I'd expect the green box to be much smaller.

**Comment by NickM-27** [comment]
Frigate will not upscale the region so it must be at least the model size (320x320 for the default model). 

In frigate 0.13 frigate keeps a database of historical detections from the timeline in a grid to know the expected sizes of objects in different parts of the image. It will then use this expected size to adjust the size of the region chosen for initial object detection

**Reply by Spectre5** [reply]
I am running 0.13 - is there a way to reset this database after making changes to detection?

And I do know there is the minimum size, but it seems to make the box even bigger than that, esp when it uses the whole frame.  I have the detect stream at 640x480 (for this camera, a reolink doorbell)

**Reply by NickM-27** [reply]
The only way is manually via sqlite, but I'm curious what changes were made to detection that you expect would require a DB reset.

**Reply by Spectre5** [reply]
Just to see what it selects for the region without any detection "memory" to understand why it expands to larger than 320x320 when it doesn't seem it should.

**Reply by NickM-27** [reply]
The best way to learn about your grid data is to view `frigate:5000/api/<camera>/grid.jpg` which will show the grid and data overlaid on the camera

**Reply by Spectre5** [reply]
Great, I'll check that out - thanks!

**Reply by charlesmunger** [reply]
If you want to reset the region history for a camera (maybe you repositioned it or replaced it with a different resolution) I did this by:

`$ sqlite3 config/frigate.db`

```
sqlite> delete from regions;
sqlite> .exit
```

If you want to delete only a single camera, you could do 

```
sqlite> delete from regions where camera = 'camera_name'
sqlite> .exit
```
replacing `camera_name` with your camera's name.

**Comment by ivanjx** [comment]
this issue prevents far objects to be detected. as you can see, the motion box is very small but the region sent to the detector (320x320) model is the entire frame size (1080p).
![image](https://github.com/user-attachments/assets/9712e526-64b7-4072-92e7-67087c24bfe3)


update: clearing the `regions` table fixed it for me. frigate really need to add a button somewhere in settings to allow us to reset it easily.

![image](https://github.com/user-attachments/assets/904618cf-6667-40f9-b579-14b1de94ed10)

**Reply by cvroque** [reply]
Recently, in order to better detect small animals I swapped my detect resolution from 640p to 1080p but detection got way worse, after a lot of trial and error I figured the region being sent to the detector was really wrong and not matching the motion boxes:
<img width="1530" height="663" alt="image" src="https://github.com/user-attachments/assets/f0e72a56-44d4-43e1-a821-dcb59c4af68a" />
I did read quite a few threads before finally ending up here! Deleting the region tables really solved the issue, thanks a lot for documenting this.

I believe there should be an easier way to clean up the region tables or maybe the existing data should be removed when the detection stream or resolution changes.

**Comment by ivanjx** [comment]
is there a way to disable the grid system here? the main problem is when a person is detected the size of course would be big but when cats or dogs pass the grid will be too big for the model to detect since it will need to be scaled down to fit the input size.

![image](https://github.com/user-attachments/assets/761a6b89-8cd6-4b08-b2da-30e6d8c6e132)

![image](https://github.com/user-attachments/assets/e117e9d4-14aa-495a-b204-3fd6694ec038)

cats passing in the middle is completely invisible to the model since the region is too big.

**Reply by NickM-27** [reply]
how often do cats pass through? this issue would indicate a disproportionate amount of large objects to small objects being detected

**Reply by ivanjx** [reply]
very rarely like once or twice a day, while `person` pass very regularly

**Reply by ivanjx** [reply]
this is the region size for further motion (same image size)

![image](https://github.com/user-attachments/assets/a4942532-3c19-465b-ade7-6cf070a90896)

**Reply by NickM-27** [reply]
what is the detect resolution?

**Reply by ivanjx** [reply]
it follows the camera (2K or 2688×1520). if i scale it to 1080p or 720p it reduces the detection for a person outside of the gate greatly. model is yolo nas with size 256x256.
